src/mmw/js/src/core/utils.js

```diff
@@ -433,6 +433,13 @@ var utils = {
             });
             return newRow;
         });
+    },
+
+    calculateVisibleRows: function(minScreenHeight, avgRowHeight, minRows) {
+        var screenHeight = document.documentElement.clientHeight;
+        return Math.max(
+            Math.floor((screenHeight - minScreenHeight) / avgRowHeight) || minRows,
```

Comment:
Hmm good catch. In my mind I was trying to prevent cases with negative values when `screenHeight` is less than `minScreenHeight`, but I see this does nothing for that. Will replace with something that works.